test/integration/shoots/operations/containerruntime.go

```diff
@@ -51,6 +51,11 @@ var _ = ginkgo.Describe("Shoot container runtime testing", func() {
 			Name:              extensionsv1alpha1.CRINameContainerD,
 			ContainerRuntimes: nil,
 		}
+		// containerD is supported only with Ubuntu OS for now.
+		// TODO: adapt/remove this when containerD is available on other OS.
+		containerdWorker.Machine.Image = &gardencorev1beta1.ShootMachineImage{
```

Comment:
Hmm, I'm not sure about that, but I think it would be better to instead skip the test if the original worker pool (the one you copy). That way, we would prevent running the test multiple times for coreos, suse and so on, but with the same stuff we test.
But I am not sure, if we actually run the complete shoot suite for every operating system. @schrodit @dguendisch Can you clarify, please?

Reply:
Generally the right approach would be to skip/fail the test when the passed OS is not the required one - kind of precondition for the test.
As far as I understood, currently there is no mechanism from tm side to restrict test to given OS, that why I am proposing this workaround.

Reply:
I think running the integration tests multiple test with different OS is an implementation detail on the TestMachinery side? Lets see what @schrodit @dguendisch say

Reply:
Yes, agreed, that's kind of what I am suggesting. I would propose to check the image name of the original worker pool and if it isn't `ubuntu` call `ginkgo.Skip()`. WDYT?

Reply:
Generally both cases are valid.
I would rather go with tim's proposal as otherwise we would run the test for all non ubuntu tests.
Because it would mean that we test containerd on ubuntu even the test flavor specifies that we would want to test only coreos or suse.

Reply:
> if we actually run the complete shoot suite for every operating system.

Yes we do. And as you've written, we currently cannot skip certain tests from TM side for different OSes. Hence I like @tim-ebert 's suggestion to check the original worker pool and and only add the containerd one when there's a chance of success and skip it otherwise.

Reply:
@ialidzhikov  WDYT?

Reply:
It makes sense. I adapted the PR to skip the test for non-ubuntu worker pool

Reply:
Thanks for clarifying :)